tool_changing_capability/src/tool_changing_capability.cpp

```diff
@@ -87,13 +87,16 @@ void ToolChangingCapability::changeEndEffectorCB(
 
 bool ToolChangingCapability::enableEndEffector(const std::string& end_effector)
 {
-  if (!robot_model_->hasEndEffector(end_effector))
+  if (!end_effector.empty() && !robot_model_->hasEndEffector(end_effector))
   {
     RCLCPP_ERROR_STREAM(LOGGER, "Unknown end-effector `" << end_effector << "`");
     return false;
   }
-  RCLCPP_INFO_STREAM(LOGGER, "Setting current end-effector to `" << end_effector << "`");
+  RCLCPP_INFO_STREAM_EXPRESSION(LOGGER, !end_effector.empty(),
+                                "Setting current end-effector to `" << end_effector << "`");
   planning_scene_monitor::LockedPlanningSceneRW scene(context_->planning_scene_monitor_);
+  // TODO(JafarAbdi): I don't think we need to cache the acm at all, we only need to set the default entry for the
+  // end-effectors' links
```

Comment:
But what about the tool that is being attached? If we don't cache ACM we'll have to make some assumptions, like allowing collisions between it and wrist. I can imagine a scenario where the tool might be allowed to collide with certain things in the ACM so caching might be appropriate to handle that.